table/block_based/filter_policy.cc

```diff
@@ -61,6 +66,20 @@ class XXPH3FilterBitsBuilder : public BuiltinFilterBitsBuilder {
     // requirements.
     if (hash_entries_.empty() || hash != hash_entries_.back()) {
       hash_entries_.push_back(hash);
+      if (cache_res_mgr_ &&
+          (hash_entries_.size() % (kUint64tHashEntryCacheResBucketSize / 2)) ==
+              0 &&
+          (hash_entries_.size() / (kUint64tHashEntryCacheResBucketSize / 2)) %
+                  2 ==
+              1) {
+        hash_entry_cache_res_bucket_handles_.emplace_back(nullptr);
+        Status s =
+            cache_res_mgr_
+                ->MakeCacheReservation<CacheEntryRole::kFilterConstruction>(
+                    kUint64tHashEntryCacheResBucketSize * sizeof(hash),
+                    &hash_entry_cache_res_bucket_handles_.back());
+        s.PermitUncheckedError();
+      }
```

Comment:
Note: compared with previous solution, bucket charging causes around +1.5ns/key for Bloom Filter and +2.5ns/key for Ribbon Filter. Should be fine (?)

Reply:
Seems fine. Thanks